airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -3086,30 +3086,19 @@ async def get_job_instance(
         with await self.service_file_as_context() as f:
             return Job(job_id=job_id, project=project_id, service_file=f, session=cast(Session, session))
 
-    async def get_job_status(
-        self,
-        job_id: str | None,
-        project_id: str | None = None,
-    ) -> str | None:
-        """
-        Polls for job status asynchronously using gcloud-aio.
-
-        Note that an OSError is raised when Job results are still pending.
-        Exception means that Job finished with errors
-        """
+    async def get_job_status(self, job_id: str | None, project_id: str | None = None):
         async with ClientSession() as s:
-            try:
-                self.log.info("Executing get_job_status...")
-                job_client = await self.get_job_instance(project_id, job_id, s)
-                job_status_response = await job_client.result(cast(Session, s))
-                if job_status_response:
+            job_client = await self.get_job_instance(project_id, job_id, s)
+            job = await job_client.get_job()
+            status = job.get("status", {})
+            if status["state"] == "DONE":
+                if "errorResult" in status:
+                    job_status = "error"
```

Comment:
Can't we just return here instead of putting it in a variable?

Reply:
I don't have strong preferences like we should return here itself i.e. in each block or store result in a variable and return at the end. before this PR we were returning at the end. maybe we can skip some checks by retuning early, right?

Reply:
see -> 06f27e3399e96b19ef197f6e3bd0f816eeaa538e